### DISCUSSION TITLE
Clarification on Blocked PRs, Code Coverage Execution, and Dependencies

### DISCUSSION THREAD

**Discussion by Rd4dev** [oppia/oppia-android#5375] [GSoC 2024] [answered]
The [project description](https://github.com/oppia/oppia/wiki/Google-Summer-of-Code-2024#41-code-coverage-support-and-enforcement) mentions that 

> "upgrading to a newer version of rules_kotlin will be done as part of #4886 ahead of GSoC coding period starting."

 Could you clarify if this upgrade will occur before GSoC starts or as part of the 4.1 project? Additionally, I'd like to know if you were able to build and obtain coverage reports by solely executing #4886. In my attempt to incorporate the new rules_kotlin, while having to upgrade Bazel to version 6.2.0 or higher, I had to test with variable dependency versions, include a few packages like [@bazel-skylib](https://github.com/bazelbuild/bazel-skylib), handle cycle dependencies, and address testonly cases. 

**Sample generated Code Coverage report:**

![Screenshot (212)](https://github.com/oppia/oppia-android/assets/122200035/43d8cb99-511d-4691-8992-372e0bd038c0)
I just wanted to confirm if obtaining Bazel code coverage and executing them via CLI will be incorporated before GSoC or as part of 4.1.

Also, considering that the PR is blocked by #4092 (add build stats CI workflow) -> #4937 (upgrade to Kotlin 1.6) -> #4931 (update Dagger structure) -> #4925 (migrate to rules_jvm_external) -> #4929 (to upgrade compute affected tests) -> #5315, would getting every PR resolved be necessary for this project's completion?

**Comment by adhiamboperes** [comment, marked as answer]
Hi @Rd4dev, the upgrades you mention are not part of the project, but work we will do to support the project. These should be done before the GSoC  coding period.

**Reply by Rd4dev** [reply]
Thank you so much @adhiamboperes for the clarification. 
I appreciate knowing that these upgrades will be completed in advance to support the 4.1 initiative.
